### PR TITLE
Recipe Service

### PR DESCRIPTION
**Pull request by MandyGan**
## Summary
- Create a recipe using spoonacular API.
- Update menu controller to create a menu using recipes created with the recipeService.

### CODE REVIEW

File: src/controllers/menu.controller.ts [hunk 1]
```diff
@@ -1,13 +1,20 @@
 import express, { Request, Response } from "express";
 import { AppDataSource } from "../config/config";
 import { Menu } from "../models/menu.entity"
+import { recipeService } from "../services/recipe.service";
+import { Recipe } from "../models/recipe.entity";
 
 const app = express();
 app.use(express.json());
 
+
```
Comment (nicorithner) [inline src/controllers/menu.controller.ts:10]: Can you please remove this extra line?

File: src/controllers/menu.controller.ts [hunk 2]
```diff
@@ -61,10 +94,14 @@ export const updateMenu = async (req: Request, res: Response) => {
     }
 }
 
+
```
Comment (nicorithner) [inline src/controllers/menu.controller.ts:97]: Another new extra blank line. Try to lint files before committing if you remember. Thank you 🙏🏻

File: src/controllers/menu.controller.ts [hunk 3]
```diff
@@ -48,10 +66,25 @@ export const createMenu = async (req: Request, res: Response) => {
...
+            relations: { recipes: true, },
+        });
+        const { name, recipes } = req.body;
         if (!menuToUpdate) throw new Error(`Menu with id of ${+req.params.id} can not be found!`)
-        await Menu.update(menuToUpdate.id, req.body);
-        const menuUpdated = await Menu.findBy({ id: +req.params.id })
+        menuToUpdate.name = name;
+        const categories = req.body.categories;
+        const menuRecipes = Array();
+        for (const category of categories) {
```
Comment (nicorithner) [inline src/controllers/menu.controller.ts:78]: This segment looks the same to the create method. Do we need to extra into a helper method?

File: src/models/menu.entity.ts [hunk 4]
```diff
@@ -1,8 +1,9 @@
 import {
-    BaseEntity, Entity, Column, PrimaryGeneratedColumn, ManyToOne, JoinColumn
+    BaseEntity, Entity, Column, PrimaryGeneratedColumn, ManyToOne, JoinColumn, ManyToMany, JoinTable
 } from "typeorm"
 
 import { User } from "./user.entity"
+import { Recipe } from "./recipe.entity"
```
Comment (nicorithner) [inline src/models/menu.entity.ts:7]: Extra blank line

File: src/routes/menu.routes.ts [hunk 5]
```diff
@@ -1,149 +1,152 @@
...
-    //Update a menu with id
-    app.put("/menus/:id", updateMenu)
+  //Update a menu with id
+  app.put("/menus/:id", updateMenu)
 
-    //Delete a menu with id
-    app.delete("/menus/:id", deleteMenu)
+  //Delete a menu with id
+  app.delete("/menus/:id", deleteMenu)
+  //Delete recipe from the menu
```
Comment (nicorithner) [inline src/routes/menu.routes.ts:148]: Here seems like that comment belong one line below and a blank line should be between it and the above route

File: src/controllers/menu.controller.ts (hunk 1 above)
Comment (MandyGan) [reply src/controllers/menu.controller.ts:17]: extra line removed!

File: src/controllers/menu.controller.ts (hunk 2 above)
Comment (MandyGan) [reply src/controllers/menu.controller.ts:100]: Checked

File: src/controllers/menu.controller.ts (hunk 3 above)
Comment (MandyGan) [reply src/controllers/menu.controller.ts:83]: Will solve this in a separate issue/

File: src/models/menu.entity.ts (hunk 4 above)
Comment (MandyGan) [reply src/models/menu.entity.ts:7]: checked

File: src/routes/menu.routes.ts (hunk 5 above)
Comment (MandyGan) [reply src/routes/menu.routes.ts]: checked

File: src/controllers/menu.controller.ts [hunk 6]
```diff
@@ -1,12 +1,19 @@
...
 import { AppDataSource } from "../config/config";
-import { Menu } from "../models/menu.entity"
+import { Menu } from "../models/menu.entity";
 import { recipeService } from "../services/recipe.service";
-import { Recipe } from "../models/recipe.entity";
 
 const app = express();
 app.use(express.json());
 
+const getRecipes = async (categories) => {
```
Comment (nicorithner) [inline src/controllers/menu.controller.ts:9]: Looks great. Thank you for making a re-usable function 🙏🏻

**Review by nicorithner — APPROVED** [review]

**Pull request merged**